Add badge to topology labels

sahil143 opened the pull request:
ODC Story: https://jira.coreos.com/browse/ODC-1192
This Pr adds the Badge to topology labels.

![image](https://user-images.githubusercontent.com/14068621/61490510-1894fe80-a97b-11e9-90d3-e74a5cf7e5d9.png)

UX:
![image](https://user-images.githubusercontent.com/14068621/61490535-29457480-a97b-11e9-876d-a487a1460aa2.png)

christianvogt commented:
@sahil143  Why are you not using `ResourceIcon` from `public/components/utils/resource-icon.tsx` and embedding it as a foreign object? You should also be calculating the amount of width it consumes so that the padding is equivalent on both sides. As seen in your two screenshots the padding is inconsistent.

christianvogt commented:
These look too different:
![image](https://user-images.githubusercontent.com/14068621/61404010-80c4e100-a8a4-11e9-97f1-d7030c63cb46.png)

`ResourceIcon` has `7px` padding and `20px` border radius.

christianvogt commented on frontend/packages/dev-console/src/components/topology/shapes/Badge.scss:
```diff
@@ -0,0 +1,11 @@
+
+.odc-badge-deployment,
+.odc-badge-deploymentconfig {
+  fill: var(--pf-global--info-color--200);
```
argh more duplication.

ResourceIcon makes use of this:
```
.co-m-resource-daemonset,
.co-m-resource-deployment,
.co-m-resource-deploymentconfig,
.co-m-resource-job,
.co-m-resource-petset,
.co-m-resource-replicaset,
.co-m-resource-replicationcontroller {
  background-color: $color-pod-overlord;
}
```

christianvogt replied:
We also support daemonsets and therefore need it's color set.

christianvogt commented on frontend/packages/dev-console/src/components/svg/SvgBoxedText.tsx:
```diff
@@ -55,24 +59,36 @@ export default class SvgBoxedText extends React.PureComponent<SvgBoxedTextProps,
           />
         )}
-        <text {...other} ref={this.textRef} x={x} y={y} textAnchor="middle" dy="0.3em">
+        {bb && kind && <Badge ref={this.badgeRef} x={x - bb.width / 2 - 2} y={y} kind={kind} />}
```
magic number `- 2` ?

sahil143 replied:
2 is the space b/w label and badge.

christianvogt commented on frontend/packages/dev-console/src/components/topology/shapes/Badge.scss:
```diff
@@ -0,0 +1,11 @@
+.odc-badge-icon {
```
Convention would be:
```suggestion
.odc-badge {
```

christianvogt commented on frontend/packages/dev-console/src/components/topology/shapes/Badge.tsx:
```diff
@@ -0,0 +1,57 @@
+  return { kindStr, KindAbbr };
+}
+
+export const Badge: React.FC<BadgeProps> = (
```
`Badge` is very generic. Maybe also call it `ResourceIcon` ?

sahil143 commented:
>These look too different:
image
ResourceIcon has 7px padding and 20px border radius

Currently, the size of the background rectangle is being calculated based on the text width. Should we make rect a fixed width?

sahil143 commented:
@christianvogt I have made the requested changes. also changed the padding and border-radius.
![Screenshot from 2019-07-18 09-49-09](https://user-images.githubusercontent.com/9278015/61428838-8cb2b080-a941-11e9-8eda-3354fad31c1a.png)

Also updated the screenshots in the description.

sahil143 commented:
/retest

christianvogt commented:
@sahil143 create a PR against your PR to try and tinker with this a bit. See https://github.com/sahil143/console/pull/1

christianvogt commented:
/lgtm
/approve

tnozicka commented:
/retest

christianvogt commented:
/lgtm

christianvogt commented:
/lgtm
